acl/aclauthz/acl_cache_test.go

```diff
@@ -0,0 +1,302 @@
+	require.NoError(t, err)
+
+	testPolicies := structs.ACLPolicies{
+		&structs.ACLPolicy{
+			ID:          "5d5653a1-2c2b-4b36-b083-fc9f1398eb7b",
+			Name:        "policy1",
+			Description: "policy1",
+			Rules:       `node_prefix "" { policy = "read" }`,
+			Syntax:      acl.SyntaxCurrent,
+			RaftIndex: structs.RaftIndex{
+				CreateIndex: 1,
+				ModifyIndex: 2,
+			},
+		},
+		&structs.ACLPolicy{
+			ID:          "b35541f0-a88a-48da-bc66-43553c60b628",
+			Name:        "policy2",
+			Description: "policy2",
+			Rules:       `agent_prefix "" { policy = "read" }`,
+			Syntax:      acl.SyntaxCurrent,
+			RaftIndex: structs.RaftIndex{
+				CreateIndex: 3,
+				ModifyIndex: 4,
+			},
+		},
+		&structs.ACLPolicy{
+			ID:          "383abb79-94ca-46c6-89b7-8ecb69046de9",
+			Name:        "policy3",
+			Description: "policy3",
+			Rules:       `key_prefix "" { policy = "read" }`,
+			Syntax:      acl.SyntaxCurrent,
+			RaftIndex: structs.RaftIndex{
+				CreateIndex: 5,
+				ModifyIndex: 6,
+			},
+		},
+		&structs.ACLPolicy{
+			ID:          "8bf38965-95e5-4e86-9be7-f6070cc0708b",
+			Name:        "policy4",
+			Description: "policy4",
+			Rules:       `service_prefix "" { policy = "read" }`,
+			Syntax:      acl.SyntaxCurrent,
+			RaftIndex: structs.RaftIndex{
+				CreateIndex: 7,
+				ModifyIndex: 8,
+			},
+		},
+	}
+
+	t.Run("Cache Miss", func(t *testing.T) {
+		authz, err := NewPolicyAuthorizerWithCache(testPolicies, cache, nil)
+		require.NoError(t, err)
+		require.NotNil(t, authz)
+
+		require.Equal(t, acl.Allow, authz.NodeRead("foo", nil))
+		require.Equal(t, acl.Allow, authz.AgentRead("foo", nil))
+		require.Equal(t, acl.Allow, authz.KeyRead("foo", nil))
+		require.Equal(t, acl.Allow, authz.ServiceRead("foo", nil))
+		require.Equal(t, acl.Default, authz.ACLRead(nil))
+	})
+
+	t.Run("Check Cache", func(t *testing.T) {
+		entry := cache.GetAuthorizer(testPolicies.HashKey())
+		require.NotNil(t, entry)
+		authz := entry.Authorizer
+		require.NotNil(t, authz)
+
+		require.Equal(t, acl.Allow, authz.NodeRead("foo", nil))
+		require.Equal(t, acl.Allow, authz.AgentRead("foo", nil))
+		require.Equal(t, acl.Allow, authz.KeyRead("foo", nil))
+		require.Equal(t, acl.Allow, authz.ServiceRead("foo", nil))
+		require.Equal(t, acl.Default, authz.ACLRead(nil))
+
+		// setup the cache for the next test
+		cache.PutAuthorizer(testPolicies.HashKey(), acl.DenyAll())
+	})
+
+	t.Run("Cache Hit", func(t *testing.T) {
+		authz, err := NewPolicyAuthorizerWithCache(testPolicies, cache, nil)
+		require.NoError(t, err)
+		require.NotNil(t, authz)
+
+		// we reset the Authorizer in the cache so now everything should be defaulted
+		require.Equal(t, acl.Deny, authz.NodeRead("foo", nil))
+		require.Equal(t, acl.Deny, authz.AgentRead("foo", nil))
+		require.Equal(t, acl.Deny, authz.KeyRead("foo", nil))
+		require.Equal(t, acl.Deny, authz.ServiceRead("foo", nil))
+		require.Equal(t, acl.Deny, authz.ACLRead(nil))
+	})
+}
+
+func TestResolveWithCache(t *testing.T) {
```

Comment:
Previously `TestStructs_ACLPolicies_resolveWithCache`, updated the test for the new function signature, but otherwise unchanged